mm2src/lp_swap.rs

```diff
@@ -264,10 +269,52 @@ async fn recv_swap_msg<T>(
 /// in order to give different and/or heavy communication channels a chance.
 const BASIC_COMM_TIMEOUT: u64 = 90;
 
+#[cfg(not(feature = "custom-swap-locktime"))]
 /// Default atomic swap payment locktime, in seconds.
 /// Maker sends payment with LOCKTIME * 2
 /// Taker sends payment with LOCKTIME
-pub const PAYMENT_LOCKTIME: u64 = 3600 * 2 + 300 * 2;
+const PAYMENT_LOCKTIME: u64 = 3600 * 2 + 300 * 2;
+
+#[cfg(feature = "custom-swap-locktime")]
+lazy_static! {
+    /// Default atomic swap payment locktime, in seconds.
+    /// Maker sends payment with LOCKTIME * 2
+    /// Taker sends payment with LOCKTIME
+    static ref PAYMENT_LOCKTIME: u64 = get_payment_locktime_from_mm2config();
+}
+
+#[cfg(feature = "custom-swap-locktime")]
+/// Reads `payment_locktime` from cli or MM2.json configuration and returns it.
+/// Returns 900 when `payment_locktime` is invalid or not provided.
+fn get_payment_locktime_from_mm2config() -> u64 {
+    let args: Vec<OsString> = env::args_os().collect();
+    let first_arg = args.get(1).and_then(|arg| arg.to_str());
+    let conf = super::get_mm2config(first_arg).unwrap();
+
+    match conf["payment_locktime"].as_u64() {
+        Some(lt) => lt,
+        None => {
+            let def = 900;
+            warn!(
+                "payment_locktime is either invalid type or not provided in the configuration or
+                MM2.json file. payment_locktime will be proceeded as {} seconds.",
+                def
+            );
+            def
+        },
+    }
+}
+
+#[inline]
+/// Returns `PAYMENT_LOCKTIME` or result of `get_payment_locktime_from_mm2config()`
+/// depended on the activation of `custom-swap-locktime` feature.
+pub fn get_payment_locktime() -> u64 {
```

Comment:
Thank you!

When `#[cfg(feature = "custom-swap-locktime")]` is enabled, PAYMENT_LOCKTIME equals to a reference, so it needs to be dereferenced there.

Otherwise, compiler will complain about that like:

```sh
error[E0308]: mismatched types
   --> mm2src/lp_swap.rs:312:5
    |
311 | pub fn get_payment_locktime() -> u64 {
    |                                  --- expected `u64` because of return type
312 |     PAYMENT_LOCKTIME
    |     ^^^^^^^^^^^^^^^^ expected `u64`, found struct `PAYMENT_LOCKTIME`
    |
help: consider dereferencing the type
    |
312 |     *PAYMENT_LOCKTIME
    |     +

For more information about this error, try `rustc --explain E0308`.
error: could not compile `mm2` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Reply:
My bad, keeping it like it is alright then 🙂